### PR TITLE
Feature/search bar

### PR DESCRIPTION
**Pull request by KristinaT**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -48,13 +67,19 @@ class App extends Component {
...
 
-      const url = `${starWarsApiUrl}?search=${item}`;
+      const url = `${Constants.STARWARS_API_URL}?search=${item}`;
       const response = await fetch(url);
       const json = await response.json();
       
-      // TODO: change logic here
       if (json.count >= 1) {
-        return Promise.resolve(json.results[0].name);
+        const resultObject={
```
Comment (filipboev) [inline src/App.js]: You can destructure this, for instance 
```js
const { name, height, mass, birth_year, gender } = json.results[0]
```
and then just create the object like so:
```js
const resultObject = {
 name,
 height, 
 mass,
 birth_year, 
 gender
}
```

File: src/App.js [hunk 2]
```diff
@@ -69,13 +94,17 @@ class App extends Component {
...
+            * Star Wars Goodness *
             <SearchBar
              inputElement= {this.inputElement}
              onChangeInput = {this.handleSearchInput}
              currentItem={this.state.searchedItem}
              addItemForSearch={this.addItemForSearch} 
               />
+            <SearchBarDetails 
+            searchedItemDetails={this.state.searchedItemDetails.text}
+            showSearchBarDetails={this.state.foundDetails}
```
Comment (filipboev) [inline src/App.js]: It's good to destructure this within the render method. Anything that requires you to repeat accessing properties on the same object more than once can be hoisted up:

```js
 const { searchedItemDetails, foundDetails } = this.state
```

File: src/Components/SearchBarDetails.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Representing the details from the searched character
+ */
+const SearchBarDetails = ({ searchedItemDetails, showSearchBarDetails }) => {
+
+    var divStyle = {
+        fontSize: '18px',
+        padding:0
+        };
```
Comment (filipboev) [inline src/Components/SearchBarDetails.js]: Create a separate JS file with common themes and put that there, you will eventually need more of those in a central place, and not in components as hardcoded values.

File: src/Components/SearchBarDetails.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    let detailItems = Object.keys(searchedItemDetails).map((key, index) => {
+        return <li key={index}> {[searchedItemDetails[key]]}</li>;
```
Comment (filipboev) [inline src/Components/SearchBarDetails.js]: If the list changes dynamically or items are to be deleted, having an index as a key is a [bad idea. ](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318)

File: src/App.js [hunk 5]
```diff
@@ -1,15 +1,25 @@
...
+          name:'',
+          height:'',
+          mass:'',
+          birthday:'',
+          gender:''
+        },
+        key: '' }
     }
     this.handleSearchInput.bind(this);
     this.addItemForSearch.bind(this);
```
Comment (filipboev) [inline src/App.js]: You don't need to ```bind``` here, those functions are already arrow functions, so they take up their parent's ```this```.

**Review by filipboev — CHANGES_REQUESTED** [review]

File: src/Components/SearchBarDetails.js (hunk 3 above)
Comment (KristinaT) [reply src/Components/SearchBarDetails.js]: Yep, I was planning to do that in the future.

File: src/Components/SearchBarDetails.js (hunk 4 above)
Comment (KristinaT) [reply src/Components/SearchBarDetails.js]: I agree, but then what should I use as a key since there isn't a unique ID in the response from the API?

Comment (filipboev) [reply src/Components/SearchBarDetails.js]: If they don't change, index is fine, if you plan to re-arrange or delete some list items, then they should be unique.

**Pull request merged**